test/Spec.hs

```diff
@@ -15,6 +15,16 @@ spec = do
             parseIExp "1+1" `shouldBe` (Right $ IExp (IExpInt 1) Plus (IExpInt 1))
             parseIExp "x-1" `shouldBe` (Right $ IExp (IExpVar $ Identifier "x") Minus (IExpInt 1))
             parseIExp "x*y" `shouldBe` (Right $ IExp (IExpVar $ Identifier "x") Mult (IExpVar $ Identifier "y"))
+            parseIExp "x/1" `shouldBe` (Right $ IExp (IExpVar $ Identifier "x") Div (IExpInt 1))
+            parseIExp "x" `shouldBe` (Right $ IExpVar $ Identifier "x")
+            parseIExp "1" `shouldBe` (Right $ IExpInt 1)
+            parseIExp "2==y" `shouldBe` (Right $ IExp (IExpInt 2) Equals (IExpVar $ Identifier "y"))
+            parseIExp "2^y" `shouldBe` (Right $ IExp (IExpInt 2) Exponent (IExpVar $ Identifier "y"))
+            -- parseIExp "\02^y" `shouldBe` (Left "failed" Message)
+
+            -- parseIExp "x/0" `shouldBe` (Right $ IExp (IExpVar $ Identifier "x") Div (IExpInt 0)) -- should be changed to error later
```

Comment:
Actually this isn't a parser level issue. Haskell actually allows division by zero to be well-typed and you get a runtime error.